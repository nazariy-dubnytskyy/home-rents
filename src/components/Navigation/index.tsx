import * as React from 'react';
import { Link } from 'react-router-dom';

import {
  Wrapper,
  Menu,
  MenuItem,
} from './styles';

function Navigation() {
  return (
    <Wrapper>
      <Menu>
        <MenuItem><Link to='/'>Home</Link></MenuItem>
        <MenuItem><Link to='/register-tenant'>Register</Link></MenuItem>
      </Menu>
    </Wrapper>
  )
}

export default Navigation;