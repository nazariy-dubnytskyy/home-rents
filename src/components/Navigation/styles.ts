import styled from 'styled-components';

export const Wrapper = styled.nav`
  padding: 20px 0;
`;

export const Menu = styled.ul`
  display: flex;
  justify-content: flex-end;
`;

export const MenuItem = styled.li`
  padding: 20px;
  font-weight: 800;
  letter-spacing: 2px;
  text-transform: uppercase;
`;