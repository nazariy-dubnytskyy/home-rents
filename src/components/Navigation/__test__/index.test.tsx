import * as React from 'react';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';

import { render, RenderResult } from '../../../utils/test-utils';
import Navigation from '..';
import { BrowserRouter } from 'react-router-dom';

let renderResult: RenderResult;

describe('Navigation component', () => {
  beforeEach(() => {
    renderResult = render(
      <BrowserRouter>
        <Navigation />
      </BrowserRouter>
    );
  });

  it('renders correctly', () => {
    const { container } = renderResult;

    expect(container.firstChild).toMatchSnapshot();
  });

  it('contains Home and Register links', () => {
    const { getByText } = renderResult;
    const homeLink = getByText('Home').closest('a');
    const registerLink = getByText('Register').closest('a');

    expect(homeLink).toHaveAttribute('href', '/');
    expect(registerLink).toHaveAttribute('href', '/register-tenant');
  });
});