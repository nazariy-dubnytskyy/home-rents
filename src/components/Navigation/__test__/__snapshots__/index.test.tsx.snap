// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component renders correctly 1`] = `
<nav
  class="sc-AxjAm dziQKn"
>
  <ul
    class="sc-AxirZ gvYqiO"
  >
    <li
      class="sc-AxiKw AaZUP"
    >
      <a
        href="/"
      >
        Home
      </a>
    </li>
    <li
      class="sc-AxiKw AaZUP"
    >
      <a
        href="/register-tenant/name"
      >
        Register
      </a>
    </li>
  </ul>
</nav>
`;
